pkg/util/net/ip/ip.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+package ip
+
+import (
+	"io/ioutil"
+	"net"
+
+	netutils "k8s.io/utils/net"
+)
+
+const (
+	// sysfs
+	disableIPv6Path = "/proc/sys/net/ipv6/conf/default/disable_ipv6"
+
+	ipv4Loopback = "127.0.0.1"
+)
+
+// GetIPZeroAddress returns INADDR_ANY or INADDR6_ANY (according sysctl disable_ipv6)
+func GetIPZeroAddress() string {
+	disableIPv6, err := readSysfsFile(disableIPv6Path)
```

Comment:
I dont see why, i added prints and checked exactly what happens,
also the only case where ipv6 is choosen is if all went well, and the value isnt "1"
i checked its indeed "0" exactly

Reply:
I do not know to what you refer, I meant unit testing this.

Reply:
readSysfsFile as you said shouldnt be used in unittest 
because its reading a sysfs file

Reply:
It can be injected as a stub in tests.
The problem here is that logic is mixed with it, therefore not testing means logic is not checked.

Code must take into account tests and needs to be designed in a testable way.
I will not block on this one, but you should note and recognize the problem.